### PR TITLE
Blog post about 1.13 release

### PR DESCRIPTION
**Pull request by flavio**
This fixes https://github.com/kubewarden/kubewarden.io/issues/227

Note: I don't like how the table is rendered by hugo. I'll try to address that.

### CODE REVIEW

**Review by fabriziosestito — APPROVED** [review]
LGTM, I think it does a great job explaining the 1.13 changes.

**Comment by flavio** [comment]
@jhkrug do you mind taking a look when you've some spare time?

File: content/blog/2024/06/kubewarden-1-13-release.md [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+idle memory consumption.
+As a pleasant side effect, we also noticed an improvement in the performances of the Policy Server thanks to the usage of jemalloc.
+
+These are some numbers we obtained while doing performance tests of a Gatekeeper policy fetching, among other resources, 10000 RoleBindings:
+
+|                    | HTTP Request Duration (avg) | Max RSS Under load | Idle RSS |
+| ------------------ | :-------------------------: | :----------------: | :------: |
+| Policy Server 1.12 |          436.15ms           |       1.4 Gb       |  1.2 Gb  |
+| Policy Server 1.13 |          233.663ms          |       1.2 Gb       |  264 Mb  |
+
```
Comment (jvanz) [inline content/blog/2024/06/kubewarden-1-13-release.md:44]: This table looks small in my screen:

image

I do not know if it is possible to make it bigger. If it is, we should do it.

**Review by jvanz — APPROVED** [review]
LGTM. I have just a comment about the table small. But I do not think that should prevent us from merging this PR

File: content/blog/2024/06/kubewarden-1-13-release.md (hunk 1 above)
Comment (flavio) [reply content/blog/2024/06/kubewarden-1-13-release.md:44]: @jordojordo can you help me fix the CSS the proper way?

I've looked a bit into that and I **think** I found the issues. The `<p>` elements of the blog have `font-size: 2.1rem;` and `line-height: 2.8rem;`, while the contents of the table have "vanilla" settings, that leads to the table's text being smaller that the rest of the article

This is how the table looks like once I change these values

image

I don't know however how to properly change that. I don't want to hardcode these values, shouldn't we instead inherit them somehow?

File: content/blog/2024/06/kubewarden-1-13-release.md [hunk 2]
```diff
@@ -0,0 +1,92 @@
+---
+title: Kubewarden 1.13 release
+authors:
+  - Flavio Castelli
+date: 2024-06-05
```
Comment (viccuad) [inline content/blog/2024/06/kubewarden-1-13-release.md:5]:
```suggestion
date: 2024-06-06
```

File: content/blog/2024/06/kubewarden-1-13-release.md [hunk 3]
```diff
@@ -0,0 +1,92 @@
...
+| Policy Server 1.12 |          436.15ms           |       1.4 Gb       |  1.2 Gb  |
+| Policy Server 1.13 |          233.663ms          |       1.2 Gb       |  264 Mb  |
+
+## Audit Scanner improvements
+
+The Audit Scanner runs regularly inside of Kubernetes clusters to determine the compliance status of its resources.
+It does that by issuing many HTTP requests against the Policy Servers. A Policy Server can be backed by multiple replicas of
+the Policy Server Pod.
+
+We got reports from users that the Audit Scanner was always talking with only one of the replicas. That caused the different issues.
```
Comment (viccuad) [inline content/blog/2024/06/kubewarden-1-13-release.md:51]:
```suggestion
We got reports from users that the Audit Scanner was always talking with only one of the replicas. This caused the following issues.
```

(or similar)

File: content/blog/2024/06/kubewarden-1-13-release.md [hunk 4]
```diff
@@ -0,0 +1,92 @@
...
+After some investigation, we found two different root causes of this issue.
+
+The first one was related to [kube-rs](https://github.com/kube-rs), the Rust library we use to interact with Kubernetes.
+The library was doing excessive allocations when building the initial state of a resource. We fixed this issue
+with [this PR](https://github.com/kube-rs/kube/pull/1494).
+With this fix we reduced the peak of memory usage caused by the creation of the initial state of a Kubernetes resource.
+
+The second issue was related with how the memory allocator of Rust works. By default, Rust uses the system allocator (glibc malloc in our case).
+It turns out the system allocator is not releasing the allocated memory back to the system when this is no longer needed.
+We changed Policy Server to use the [jemalloc](http://jemalloc.net/) allocator which, on top of providing great performances, it has many
```
Comment (jhkrug) [inline content/blog/2024/06/kubewarden-1-13-release.md:30]:
```suggestion
We changed Policy Server to use the [jemalloc](http://jemalloc.net/) allocator which, on top of providing great performance, has many
```

File: content/blog/2024/06/kubewarden-1-13-release.md [hunk 5]
```diff
@@ -0,0 +1,92 @@
...
+configuration knobs.
+After some experimentations, we found the right balance between performances and memory usage.
```
Comment (jhkrug) [inline content/blog/2024/06/kubewarden-1-13-release.md:32]: experimentation or experiments and singular performance

File: content/blog/2024/06/kubewarden-1-13-release.md [hunk 6]
```diff
@@ -0,0 +1,92 @@
...
+These are some numbers we obtained while doing performance tests of a Gatekeeper policy fetching, among other resources, 10000 RoleBindings:
+
+|                    | HTTP Request Duration (avg) | Max RSS Under load | Idle RSS |
+| ------------------ | :-------------------------: | :----------------: | :------: |
+| Policy Server 1.12 |          436.15ms           |       1.4 Gb       |  1.2 Gb  |
+| Policy Server 1.13 |          233.663ms          |       1.2 Gb       |  264 Mb  |
+
+## Audit Scanner improvements
+
+The Audit Scanner runs regularly inside of Kubernetes clusters to determine the compliance status of its resources.
```
Comment (jhkrug) [inline content/blog/2024/06/kubewarden-1-13-release.md:47]:
```suggestion
The Audit Scanner runs regularly inside Kubernetes clusters to determine the compliance status of its resources.
```

File: content/blog/2024/06/kubewarden-1-13-release.md [hunk 7]
```diff
@@ -0,0 +1,92 @@
...
+It does that by issuing many HTTP requests against the Policy Servers. A Policy Server can be backed by multiple replicas of
+the Policy Server Pod.
+
+We got reports from users that the Audit Scanner was always talking with only one of the replicas. That caused the different issues.
+First, the audit scan was slow. Second, when running with multiple replicas, one of the Policy Server Pods was overloaded with requests,
+while the other ones were basically idle.
```
Comment (jhkrug) [inline content/blog/2024/06/kubewarden-1-13-release.md:53]:
```suggestion
while the other ones were idle.
```

File: content/blog/2024/06/kubewarden-1-13-release.md (hunk 1 above)
Comment (viccuad) [reply content/blog/2024/06/kubewarden-1-13-release.md:44]: To me, it should go into `assets/sass/blog-main.scss` IIUC.

File: content/blog/2024/06/kubewarden-1-13-release.md [hunk 8]
```diff
@@ -0,0 +1,92 @@
...
+This issue was caused by how the Go HTTP client works. By default, the Go HTTP client uses a connection pool to reuse HTTP connections.
+Once we turned off this behavior we immediately saw the requests being spread between the different replicas of the Policy Server.
+
+Note: the Audit Scanner accesses the Policy Server by going through the `ClusterIP` of the Service that exposes that Policy Server internally.
+The `kube-proxy` component is responsible for load balancing the requests between the different Pods of the Policy Server.
+
+Once this limitation was removed, we saw a significant reduction of the processing time of the Audit Scanner, especially on clusters with many policies and
+resources to evaluate.
+
+However, not all the operations done by the Audit Scanner were doing in parallel. Starting from this release, all the operations done by the Audit Scanner
```
Comment (jhkrug) [inline content/blog/2024/06/kubewarden-1-13-release.md:64]:
```suggestion
However, not all the operations done by the Audit Scanner were happening in parallel. Starting from this release, all the operations done by the Audit Scanner
```

**Review by jhkrug — CHANGES_REQUESTED** [review]
A few minor things.

File: content/blog/2024/06/kubewarden-1-13-release.md (hunk 5 above)
Comment (flavio) [reply content/blog/2024/06/kubewarden-1-13-release.md:32]: done, thanks

File: content/blog/2024/06/kubewarden-1-13-release.md (hunk 1 above)
Comment (flavio) [reply content/blog/2024/06/kubewarden-1-13-release.md:44]: I've fixed it with abc8fe7e6960b24265d2d21583f8d36f7d02bfa4

@jordon: please take a look at that once you're back

**Pull request merged**